src/main/kotlin/turku/forge/babyproject/api/RootApiController.kt

```diff
@@ -1,19 +1,69 @@
 package turku.forge.babyproject.api
 
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonInclude.Include
+import org.springframework.hateoas.LinkRelation
+import org.springframework.hateoas.server.core.Relation
+import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder
+import org.springframework.hateoas.server.mvc.linkTo
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.CrossOrigin
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.RestController
+import turku.forge.babyproject.config.STOMP_ENDPOINT
+import turku.forge.babyproject.resources.Resource
 
-const val API_PATH = "/"
+const val API_PATH = ""
+
+// @TODO: maybe move these relations to their own ENUM if needed
+const val WS_RELATION = "ws:connect"
+const val CHANNEL_RELATION = "channel"
+const val INDEX_RELATION = "index"
 
 /**
  * Rest control for the root of the api.
  * if you serve this api under `api.example.com` all the request going to `/`
  * will be routed to this controller.
+ * @TODO: Figure out how to handle the CORS logic
  */
 @RestController
+@CrossOrigin( origins = ["*"])
 class RootApiController {
     @GetMapping(API_PATH)
-    fun index(): String {
-        return "Hello World"
+    fun index(): ResponseEntity<IndexResource> {
+        val resource = IndexResource()
+        resource.add(WebMvcLinkBuilder.linkTo(RootApiController::class.java).withSelfRel())
+        resource.add(WebMvcLinkBuilder.linkTo(RootApiController::class.java)
+                .slash(STOMP_ENDPOINT)
+                .withRel(WS_RELATION))
+
+        val channelRelation = LinkRelation.of(CHANNEL_RELATION)
+        val channelCollection: Collection<ChannelResource> = getChannels().map {
+            val link = linkTo<ChannelController> { message(it, null) }
+            val channelResource = ChannelResource(it, link.toUriComponentsBuilder().build().path)
+            channelResource.add(link.withSelfRel())
+        }
+        resource.add(channelRelation, channelCollection)
+
+        return ResponseEntity(resource, HttpStatus.OK);
+    }
+
+    private fun getChannels(): List<String> {
+        // @TODO: remove the hard coded channel names once we have a database and a way to create and store channels
+        return listOf("general", "random")
     }
 }
+
+/**
+ * This is the index resource for this file
```

Comment:
I don't quite follow what this comment is. Is it a TODO or documentation?

Reply:
Its supose to be initial documentatsion. I need to improve all these docs for all of these. Need to figure out what to write there to. Its hard to document this stuff since I need a good location for all these. 

could essentaly document all the resouces as 
```
/**
 * A <type> resource
 * @see {path.to.base.resource.package}
 */
```

That way there is one location where I can explain what a resouce is.